pycyphal/transport/can/media/pythoncan/_pythoncan.py

```diff
@@ -313,8 +337,17 @@ def _read_batch(self) -> typing.List[typing.Tuple[Timestamp, Envelope]]:
             msg = self._bus.recv(0.0 if batch else self._MAXIMAL_TIMEOUT_SEC)
             if msg is None:
                 break
-            timestamp = Timestamp.now()  # TODO: use accurate timestamping
-            loopback = False  # TODO: no possibility to get real loopback yet
+
+            if self._bus_options.hardware_timestamp:
+                timestamp = Timestamp(system_ns=time.time_ns(), monotonic_ns=Timestamp._second_to_ns(msg.timestamp))
```

Comment:
Uh-oh. Should we make `_second_to_ns` public or should we manually multiply the timestamp by 1e9 here? Reaching for a private method like this gives me pain.

Reply:
I'm not too familiar with python and thought it is a package-private method. Neither python nor linter complained though :D I think multiplying by 1e9 should be clear enough

Reply:
I am also wondering why the linter didn't complain. I am used to PyCharm yelling at me when I do something like this so I didn't even check if PyLint is configured to report issues of this kind.